crates/cargo-util-schemas/src/core/partial_version.rs

```diff
@@ -163,6 +159,22 @@ impl<'de> serde::Deserialize<'de> for PartialVersion {
     }
 }
 
+/// Error parsing a [`PartialVersion`].
+#[derive(Debug, thiserror::Error)]
+pub enum PartialVersionError {
+    #[error("unexpected version requirement, expected a version like \"1.32\"")]
+    VersionReq,
+
+    #[error("unexpected prerelease field, expected a version like \"1.32\"")]
+    Prerelease,
+
+    #[error("unexpected build field, expected a version like \"1.32\"")]
+    BuildMetadata,
```

Comment:
These are odd variants because the reason they show up in `PartialVersion` is completely different than `RustVersion`, coupling them together when there might be situations we might want them to drift apart (ie false code sharing).

Reply:
Pulled out a new `RustVersionError` type